ui/app/clinical/consultation/controllers/dispositionController.js

```diff
@@ -32,9 +32,13 @@ angular.module('bahmni.clinical')
             });
         };
 
-        var findAction = function (dispositions, action) {
-            var undoDischarge = _.find(dispositions, action);
-            return undoDischarge || {'name': ''};
+        var pushDispositionAction = function (finalDispositionActions, dispositions, action) {
+            var copyOfFinalDispositionActions = _.cloneDeep(finalDispositionActions);
+            var dispositionPresent = _.find(dispositions, action);
+            if (dispositionPresent) {
+                copyOfFinalDispositionActions.push(dispositionPresent);
+                return copyOfFinalDispositionActions;
```

Comment:
Hi @maharjunm ,
I think this `return` will be outside of `if` block. As if **dispositionPresent** is false, ** finalDispositionActions** will be `undefined`